examples/large_files/test/tests.ts

```diff
@@ -0,0 +1,153 @@
+        generateTest(origin, 'test2000001B', 'auto'),
+        //      General Cases
+        generateTest(origin, 'test1KiB', 'auto'),
+        generateTest(origin, 'test10KiB', 'auto'),
+        generateTest(origin, 'test100KiB', 'auto'),
+        generateTest(origin, 'test1MiB', 'auto'),
+        generateTest(origin, 'test10MiB', 'auto'),
+        generateTest(origin, 'test100MiB', 'auto'),
+        generateTest(origin, 'test250MiB', 'auto'),
+        generateTest(origin, 'test1GiB', 'auto'),
+        // Manual Upload
+        {
+            name: 'test manual upload',
+            test: async () => {
+                await uploadFiles('backend', [
+                    ['assets/manual/test150MiB', 'assets/test150MiB']
+                ]);
+
+                await new Promise((resolve) => setTimeout(resolve, 10 * 1000));
+
+                const response = await fetch(
+                    `${origin}/exists?path=assets/test150MiB`
+                );
+
+                return { Ok: (await response.json()) === true };
+            }
+        },
+        generateTest(origin, 'test150MiB', 'manual')
+    ];
+}
+
+/**
+ * Generate a test for file uploading. Hashes the local file and compares it to
+ * the hash of the uploaded file. Assumes that all of the files both on the
+ * canister and local side are in a directory called "assets". The parameter
+ * localDir allows for difference between the canisterPath and localPath and
+ * will be inserted between "assets" and the rest of the file path to the local
+ * asset. If localPath is defined it will be used for the localPath. Otherwise
+ * it will be assumed that the canisterPath is the same as the localPath.
+ * @param origin
+ * @param canisterPath
+ * @param localDir
+ * @param localPath
+ * @returns
+ */
+function generateTest(
+    origin: string,
+    canisterPath: string,
+    localDir?: string,
+    localPath?: string
+): Test {
+    return {
+        name: `upload: ${canisterPath}`,
+        test: async () => {
+            const canisterFilePath = join('assets', canisterPath);
+            const localFilePath = join(
+                'assets',
+                localDir ?? '',
+                localPath ?? canisterPath
+            );
+
+            const actor = await createGetFileHashActor(
+                getCanisterId('backend')
+            );
+
+            const expectedHash = (await hashFile(localFilePath)).toString(
+                'hex'
+            );
+
+            const response = await fetch(
+                `${origin}/exists?path=${canisterFilePath}`
+            );
+            const exits = await response.json();
+
+            if (exits === false) {
+                return {
+                    Err: `File ${canisterFilePath} failed to upload`
+                };
+            }
+
+            const hash = await actor.get_file_hash(`${canisterFilePath}`);
+            return { Ok: hash === expectedHash };
+        }
+    };
+}
+
+async function createGetFileHashActor(
```

Comment:
We need to add authentication to the hashing endpoint

Reply:
done